arduino-ide-extension/src/browser/contributions/interface-scale.ts

```diff
@@ -0,0 +1,233 @@
+import { inject, injectable } from '@theia/core/shared/inversify';
+import {
+  Contribution,
+  Command,
+  MenuModelRegistry,
+  KeybindingRegistry,
+} from './contribution';
+import { ArduinoMenus, PlaceholderMenuNode } from '../menu/arduino-menus';
+import {
+  CommandRegistry,
+  DisposableCollection,
+  MaybePromise,
+  nls,
+} from '@theia/core/lib/common';
+
+import { Settings } from '../dialogs/settings/settings';
+import { MainMenuManager } from '../../common/main-menu-manager';
+import debounce = require('lodash.debounce');
+
+@injectable()
+export class InterfaceScale extends Contribution {
+  @inject(MenuModelRegistry)
+  private readonly menuRegistry: MenuModelRegistry;
+
+  @inject(MainMenuManager)
+  protected readonly mainMenuManager: MainMenuManager;
+
+  private readonly menuActionsDisposables = new DisposableCollection();
+  private fontScalingEnabled: InterfaceScale.FontScalingEnabled = {
+    increase: true,
+    decrease: true,
+  };
+
+  private currentScale: InterfaceScale.ScaleSettings;
+  private currentSettings: Settings;
```

Comment:
Can you explain why IDE2 needs to duplicate the `Settings` state twice?

Reply:
One duplicate is needed for debouncing purposes, but you're right, I can achieve the same with only one variable. Changing it.

Reply:
Thank you!

Please note `isEnabled: () => this.fontScalingEnabled.increase,` and `isEnabled: () => this.fontScalingEnabled.decrease,` have absolutely no effect.

Exactly that's why you're using the `PlaceholderMenuNode` menu item to have it there in a disabled state.

Reply:
I think you're wrong here. Those lines are not meant to disable/enable the menu item, but the related commands, so that it prevents zooming in/out with shortcuts when exceeding limits.

Reply:
- The increase/decrease font size will trigger `this.updateFontSize('decrease'|'increase')`;
 - The `updateFontSize` method min/max the font size. See [here](https://github.com/arduino/arduino-ide/pull/1502/files#diff-e85455776c63a56847b610b7e290b72e1ffb4bcfb0188d3351d9fe17866c33a2R138-R154).

Where did I overlook your code? Thanks!